recipe/remove-macos-apps.patch

```diff
@@ -0,0 +1,226 @@
+diff --git a/tests/conftest.py b/tests/conftest.py
+index 4dc3a91..f546b23 100644
+--- a/tests/conftest.py
++++ b/tests/conftest.py
+@@ -16,7 +16,7 @@ menuinst_pkg_specs = [
+         "conda-test/label/menuinst-tests::package_1",
+         {
+             "win32": "Package 1/A.lnk",
+-            "darwin": "A.app/Contents/MacOS/a",
++            "darwin": "A.app",
+             "linux": "package-1_a.desktop",
+         },
+     ),
+diff --git a/tests/test_main.py b/tests/test_main.py
+index fb658a0..3337193 100644
+--- a/tests/test_main.py
++++ b/tests/test_main.py
+@@ -177,6 +177,14 @@ _pkg_specs_params = pytest.mark.parametrize(
+ @_pkg_specs_params
+ def test_menuinst_conda(tmp_path: Path, pkg_spec: str, shortcut_path: dict[str, str]):
+     "Check 'regular' conda can process menuinst JSONs"
++
++    def _clean_macos_apps(shortcuts: list[Path]):
++        if not sys.platform == "darwin":
++            return
++        for shortcut in shortcuts:
```

Comment:
Minor suggestion to add a try block here in case removal of the shortcut fails.

Reply:
If the shortcut removal fails, the test (and subsequent) tests will also fail though. That's a real issue then.

Reply:
Ahh, so we only need to verify the shortcut removals so the subsequent tests don't fail?

Reply:
Having an exception here when it fails to remove is actually a good thing at the end, too: we would be able to distinguish between permission errors and whether the menuinst is the problem.

Reply:
> Ahh, so we only need to verify the shortcut removals so the subsequent tests don't fail?

Correct. The context of this patch is that a past test on `osx-64` failed. `pytest` didn't clean up that app and then subsequent builds failed because of that left-over file. This is highly specific to macOS because `menuinst` refuses to overwrite an existing app.

Reply:
Thanks for the context!

Reply:
Nice work!

Reply:
I am a little curious what happens if there is a permission error and how that will affect the test...

Reply:
The test will just fail and the permission errors would have to be investigated. You then either can't run the test on your system or it points to a `conda-standalone`, `menuinst`, or `conda` problem that needs to be fixed.

Reply:
Ahh, I see how the try block could possibly narrow that investigation down a bit, if need be.